tiledb/cloud/taskgraphs/client_executor/udf_node.py

```diff
@@ -46,20 +67,85 @@ def task_id(self, timeout: Optional[float] = None) -> Optional[uuid.UUID]:
         self.wait(timeout)
         return self._task_id
 
+    def _is_local(self) -> bool:
+        return bool(self._environment.get("run_client_side"))
+
     def _exec_impl(
         self, parents: Dict[uuid.UUID, _base.Node], input_value: Any
     ) -> None:
         assert input_value is _base.NOTHING
 
+        if self._is_local():
+            self._exec_local(parents)
+            # TODO: Add a way that users can specify that they want to retry
+            # a local UDF remotely.
+            return
+        self._exec_remote(parents)
```

Comment:
That does make sense for now; I think the original way I was doing it was from a time when there was more logic and it would have made sense to fall through to `_exec_remote` based if `_exec_local` failed. Since we no longer do that, a regular if-else is indeed better.